document 'required' on custom fields (#20813)
opf/openproject: docs/system-admin-guide/custom-fields/README.md

1. [Activate this custom field for projects under the *Projects* tab](#add-a-custom-field-to-one-or-multiple-projects)
2. [Assign a custom field to a work package type](../manage-work-packages/work-package-types/#work-package-form-configuration-enterprise-add-on) 

> ## 'required' custom fields

Reviewer: ```suggestion
## 'Required' custom fields
```

Author: Worked in in [3dab608](https://github.com/opf/openproject/pull/20813/commits/3dab608a149c79ad8f17a7aa04cf4469716a02ba)

> Custom fields can be configured as 'required'. As the name suggests, 'required' custom fields need to be filled out when a resource like a work package or a project is created. Without the user filling in a field, the user will be prevented from creating the resource.

Reviewer: ```suggestion
Custom fields can be configured as 'required'. As the name suggests, 'required' custom fields need to be filled out when a resource like a work package or a project is created. Without filling in a field, the user will be prevented from creating the resource.
```

Author: Worked in in [3dab608](https://github.com/opf/openproject/pull/20813/commits/3dab608a149c79ad8f17a7aa04cf4469716a02ba)

> The 'required' flag can also be set on existing custom fields. That poses a problem for resources that existed before the flag was switched to being required. E.g. a work package might have already been created and worked with for some time. But because the field was not required before, it might not have been filled in. Once the custom field becomes 'required', the next user wanting to modify the work package needs to fill in the field. There are other scenarios where this might be necessary, among which are the activation of a required custom field for a project or a work package type.

Reviewer: ```suggestion
The 'required' flag can also be set on existing custom fields. This might lead to a problem for resources that existed before the flag was later set to required. For example, a work package might have already been created and worked with for some time. But because the field was not required before, it might not have been filled in. Once the custom field becomes 'required', the next user wanting to modify the work package *needs* to fill in the field. There are other scenarios where this might occur, for example when the activation of a required custom field for a project or a work package type.
```

Author: Worked in in [3dab608](https://github.com/opf/openproject/pull/20813/commits/3dab608a149c79ad8f17a7aa04cf4469716a02ba)

> In the above scenario of a required custom field being absent on existing resources, OpenProject will try to minimize the times in which users are blocked from performing their intended action. A user might want to change the name of the project with no value filled in for a required custom field. Since the UI for changing the name does not display the custom fields, the user cannot provide a value at this place and OpenProject will not enforce the 'required' setting then.

Reviewer: ```suggestion
In the above scenario of a required custom field being absent on existing resources, OpenProject will try to minimize the times in which users are blocked from performing their intended action. A user might want to change the name of the project with no value filled in for a required custom field. Since the interface for changing the name does not display the custom fields, the user cannot provide a value at this point and OpenProject will not enforce the 'required' setting then.
```

Author: Worked in in [3dab608](https://github.com/opf/openproject/pull/20813/commits/3dab608a149c79ad8f17a7aa04cf4469716a02ba)

Generally speaking, OpenProject will enforce the 'required' characteristic whenever the UI displays the field, but allows modifications when it does not. When creating a new resource, it will be enforced.

> [!NOTE]

> > Prior to version 16.6, required custom fields would always be enforced also on existing resources. This led to some actions becoming impossible or unnecessarily hard to perform which is why the behavior was adapted.

Reviewer: ```suggestion
> Prior to version 16.6, required custom fields were always enforced, even on existing resources. This led to some actions becoming impossible or unnecessarily hard to perform, which is why the behavior was adjusted.
```

Author: Worked in in [3dab608](https://github.com/opf/openproject/pull/20813/commits/3dab608a149c79ad8f17a7aa04cf4469716a02ba)
